frontend/src/tests/HealthCheck.test.tsx

```diff
@@ -0,0 +1,25 @@
+import { vi, describe, expect, it } from 'vitest';
+import { render, screen, waitFor } from '@testing-library/react';
+import axios from 'axios';
+import HealthCheck from '../components/HealthCheck';
+
+// Mock the Axios module
+vi.mock('axios');
+
+describe('HealthCheck Component', () => {
+  it('fetches and displays HealthCheck Component response data', async () => {
+    // Mocked data returned by the Axios Get request
+    const response = 'NMP API is healthy and ready!';
+
+    // Setup the mock to return the mocked data
+    vi.mocked(axios.get).mockResolvedValueOnce({ data: response });
+
+    // Render the component
+    render(<HealthCheck />);
```

Comment:
Comments like this aren't valuable. 
Of course it renders the component, it's a function called `render` taking a component as an argument :)

Reply:
Sounds good, I'll re-evaluate my comments and add what's needed, but regarding the dependencies, I don't think I will be commenting them inline but rather just add my dependencies to wiki and add description on what it's importing in the project itself. 😄

Reply:
Adding a super condensed write up about testing in your frontends readme is valuable too :) 
Just a simple 
```
Testing
The Frontend uses x,y,z for testing
you can run the tests by spinning a top while doing a trick with a yoyo
more information about testing can be found at link/to/wiki/page
```